.github/workflows/build.yml

```diff
@@ -0,0 +1,71 @@
+name: build
+
+on:
+  push:
+    branches: [ master ]
+    tags: [ v* ]
+  pull_request:
+    branches: [ master ]
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ macos-latest, ubuntu-latest, windows-latest ]
+        include:
+          - os: macos-latest
+            os_name: macos
+          - os: ubuntu-latest
+            os_name: linux
+          - os: windows-latest
+            os_name: windows
+
+    steps:
+
+    - name: Checkout code
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: Setup .NET Core SDK
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: 3.1.302
+
+    - name: Install dependencies
+      run: dotnet restore
+
+    - name: Build
+      run: dotnet build --configuration Release --no-restore
+
+    - name: Test
+      run: |
+        dotnet test --no-restore --verbosity normal -c Release  ./tests/JustSaying.UnitTests/JustSaying.UnitTests.csproj
+        dotnet test --no-restore --verbosity normal -c Release ./tests/JustSaying.Extensions.DependencyInjection.StructureMap.Tests/JustSaying.Extensions.DependencyInjection.StructureMap.Tests.csproj
+    
+    - name: Integration Test
+      if: matrix.os_name == 'linux'
+      run: |
+        goaws_tag=latest
+        docker pull pafortin/goaws:$goaws_tag
+        docker run -d --name goaws -p 4100:4100 pafortin/goaws:$goaws_tag
+        export AWS_SERVICE_URL="http://localhost:4100"
+        dotnet test --no-restore --verbosity normal ./tests/JustSaying.IntegrationTests/JustSaying.IntegrationTests.csproj
+    
+    - name: Pack
+      run: |
+        dotnet pack --no-restore --verbosity normal -c Release  --output ./artifacts/packages ./src/JustSaying/JustSaying.csproj
+        dotnet pack --no-restore --verbosity normal -c Release  --output ./artifacts/packages ./src/JustSaying.Models/JustSaying.Models.csproj
+        dotnet pack --no-restore --verbosity normal -c Release  --output ./artifacts/packages ./src/JustSaying.Extensions.DependencyInjection.Microsoft/JustSaying.Extensions.DependencyInjection.Microsoft.csproj
+        dotnet pack --no-restore --verbosity normal -c Release  --output ./artifacts/packages ./src/JustSaying.Extensions.DependencyInjection.StructureMap/JustSaying.Extensions.DependencyInjection.StructureMap.csproj
+
+    - name: Publish NuGet packages
+      uses: actions/upload-artifact@v1
+      with:
+        name: packages-${{ matrix.os_name }}
+        path: ./artifacts/packages
```

Comment:
I'll set up the secret a bit later today, then you can add the publishing part. There's also a fix I need to add to httpclient-interception I need to add for that before you copy it over.

Reply:
Here's the fix: https://github.com/justeat/httpclient-interception/pull/234

Reply:
And I've added the `NUGET_TOKEN` secret.